frontend/web/src/assets/lib/jsc3d/jsc3d.webgl.js

```diff
@@ -0,0 +1,1202 @@
+		case 'textureflat':
+		case 'texturesmooth':
+		default:
+			gl.bindBuffer(gl.ARRAY_BUFFER, mesh.compiled.coords);
+			gl.vertexAttribPointer(this.programs.picking.attributes['a_position'], 3, gl.FLOAT, false, 0, 0);
+			gl.drawArrays(gl.TRIANGLES, 0, mesh.compiled.coordCount);
+			break;
+		}
+	}
+};
+
+/**
+ * Compile a mesh according to the given render mode, generating the WebGL dependent stuff.
+ * @private
+ */
+JSC3D.WebGLRenderBackend.prototype.compileMesh = function(mesh, renderMode) {
+	if(mesh.isTrivial())
+		return false;
+
+	renderMode = mesh.renderMode || renderMode;
+
+	function makeWireframe(ibuf, vbuf, numOfFaces, trianglesOnly) {
+		var edges;
+
+		var v0, v1, v2;
+		if(trianglesOnly) {
+			edges = new Float32Array(18 * numOfFaces);
+			for(var i=0, e=0; i<ibuf.length; i+=4, e+=18) {
+				v0 = 3 * ibuf[i    ];
+				v1 = 3 * ibuf[i + 1];
+				v2 = 3 * ibuf[i + 2];
+
+				// v0 <-> v1
+				edges[e     ] = vbuf[v0    ];
+				edges[e +  1] = vbuf[v0 + 1];
+				edges[e +  2] = vbuf[v0 + 2];
+				edges[e +  3] = vbuf[v1    ];
+				edges[e +  4] = vbuf[v1 + 1];
+				edges[e +  5] = vbuf[v1 + 2];
+				// v1 <-> v2
+				edges[e +  6] = vbuf[v1    ];
+				edges[e +  7] = vbuf[v1 + 1];
+				edges[e +  8] = vbuf[v1 + 2];
+				edges[e +  9] = vbuf[v2    ];
+				edges[e + 10] = vbuf[v2 + 1];
+				edges[e + 11] = vbuf[v2 + 2];
+				// v2 <-> v0
+				edges[e + 12] = vbuf[v2    ];
+				edges[e + 13] = vbuf[v2 + 1];
+				edges[e + 14] = vbuf[v2 + 2];
+				edges[e + 15] = vbuf[v0    ];
+				edges[e + 16] = vbuf[v0 + 1];
+				edges[e + 17] = vbuf[v0 + 2];
+			}
+		}
+		else {
+			edges = [];
+			for(var i=0, j=0; i<numOfFaces; i++) {
+				v0 = 3 * ibuf[j++];
+				v1 = v0;
+				while(ibuf[j] > 0) {
+					v2 = 3 * ibuf[j++];
+					edges.push( vbuf[v1], vbuf[v1 + 1], vbuf[v1 + 2], vbuf[v2], vbuf[v2 + 1], vbuf[v2 + 2] );
+					v1 = v2;
+				}
+				j++;
+				// close the polygon
+				edges.push( vbuf[v1], vbuf[v1 + 1], vbuf[v1 + 2], vbuf[v0], vbuf[v0 + 1], vbuf[v0 + 2] );
+			}
+			edges = new Float32Array(edges);
+		}
+
+		return edges;
+	}
+
+	var gl = this.gl;
+
+	var needFlat = (renderMode == 'flat') || (renderMode == 'textureflat');
+
+	var hasTrianglesOnly = mesh.indexBuffer.length == 4 * mesh.faceCount;
+	var hasTextureCoords = mesh.texCoordBuffer && mesh.texCoordBuffer.length >= 2;
+
+	var ibuf  = mesh.indexBuffer;
+	var vbuf  = mesh.vertexBuffer;
+	var tbuf  = mesh.texCoordBuffer;
+	var tibuf = mesh.texCoordIndexBuffer || ibuf;
+	var nbuf  = mesh.vertexNormalBuffer;
+	var nibuf = mesh.vertexNormalIndexBuffer || ibuf;
+	var fnbuf = mesh.faceNormalBuffer;
+	var numOfFaces = mesh.faceCount;
+
+	if (!mesh.compiled) {
+		/*
+		 * Rebuild all primitives from scratch.
+		 */
+
+		mesh.compiled = {
+			isIndexed: false
+		};
+
+		mesh.compiled.pickingId = new Float32Array([
+			(mesh.internalId & 0xff0000) / 16777216, (mesh.internalId & 0xff00) / 65536, (mesh.internalId & 0xff) / 256
+		]);
+
+		var triangles, edges, coords, normals, texcoords;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'triangles' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AUnusedVariable)